### PR TITLE
feat(api): Add G-Code Parser Makefile Execution

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Added support for running the G-Code Parser CLI through the `api` project Makefile. 

# Changelog

* Add device name to text mode output
* Add HTML formatting for GCodeDiffer HTML output
* Add strings_are_equal method to GCodeDiffer
* Utilize strings_are_equal method in cli.py
* Add Makefile targets to run G-Code Parser
* Add first protocol test to Makefile

# Review requests

Check if I am doing anything dumb in the Makefile

# Risk assessment

None

### CODE REVIEW

File: api/Makefile [hunk 1]
```diff
@@ -194,30 +194,127 @@ term:
...
+
+# open-diff
+#	Description - Helper function to open diff file in google-chrome
+#	Example 1 - Open c
+#		open-diff
+#	Example 2 - Open /path/to/file.html
+#		open-diff diff_file=/path/to/file.html
+.PHONY: open-diff
+open-diff:
+	google-chrome file://$(diff_file)
```
Comment (amitlissack) [inline api/Makefile]: Does this have to bee `google-chrome`? Would `open` launch the default browser?

**Review by amitlissack — COMMENTED** [review]
Code looks good to me. I'll test it out before +1.

**Review by amitlissack — APPROVED** [review]
Tested `make g-code-smoothie-protocol-run` and worked fine. 

I recommend someone who is more Makefile knowledgable approves before merging.

File: api/Makefile (hunk 1 above)
Comment (DerekMaggio) [reply api/Makefile]: I don't have open on my system :/

File: .github/workflows/g-code-confirm.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  workflow_dispatch:
+
+jobs:
+  confirm-g-code:
+    # Please trigger
+    name: 'confirm G-Code'
+    runs-on: 'ubuntu-18.04'
+    steps:
+      - uses: 'actions/checkout@v2'
```
Comment (sotaoverride) [inline .github/workflows/g-code-confirm.yaml:26]: im trying to find this action under actions for this repo? is that where its picked up from?

File: .github/workflows/g-code-confirm.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+name: G-Code-Confirm
+
+on:
+  # Run on any change to the api directory
+  pull_request:
```
Comment (sotaoverride) [inline .github/workflows/g-code-confirm.yaml:5]: maybe set it to trigger for all branches for now too?

**Review by sotaoverride — COMMENTED** [review]
trying to figure if you need to add actions to this branch?

File: .github/workflows/g-code-confirm.yaml (hunk 2 above)
Comment (sotaoverride) [reply .github/workflows/g-code-confirm.yaml:26]: or are these some gh actions you can add straight away ??

**Review by sotaoverride — COMMENTED** [review]
do new workflows just get picked up, or you have to add them the first time around?https://github.com/Opentrons/opentrons/actions/new

File: .github/workflows/g-code-confirm.yaml (hunk 2 above)
Comment (sotaoverride) [reply .github/workflows/g-code-confirm.yaml:26]: figured these actions just come with github..

File: .github/workflows/g-code-confirm.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+name: G-Code-Confirm
```
Comment (sotaoverride) [inline .github/workflows/g-code-confirm.yaml:1]: missing quotes here, 'G-Code-Confirm'

**Review by sotaoverride — DISMISSED** [review]
think you just had the quotes missing for you workflow name..

**Review by sotaoverride — COMMENTED** [review]
there were a few annoying yaml syntax things - ran the gavi workflow validator. https://github.com/Opentrons/opentrons/blob/test-ghactions/.github/workflows/g-code-confirm.yaml

File: .github/workflows/g-code-confirm.yaml (hunk 4 above)
Comment (sotaoverride) [reply .github/workflows/g-code-confirm.yaml:1]: take that back, apparently its valid to have no quotes??

File: .github/workflows/g-code-confirm.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+on:
+  # Run on any change to the api directory
+  pull_request:
+    paths:
+      -'*'
+  push:
+    paths:
+      -'*'
+
+  workflow_dispatch:
```
Comment (sotaoverride) [inline .github/workflows/g-code-confirm.yaml:18]: not sure how this works - did a quick read and makes me think it needs a webhook to start or something? maybe thats why this workflow isnt running when a push / pr happens ?https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event

**Review by Laura-Danielle — APPROVED** [review]
lgtm

**Pull request merged**